### PR TITLE
IBX-6485:  Forced `RichText\Renderer`to utilize `PermissionResolver` instead of `AuthorizationChecker`

### PR DESCRIPTION
**Pull request by barw4**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | [IBX-6485](https://issues.ibexa.co/browse/IBX-6485)
| **Improvement**| yes
| **New feature**    | no
| **Target version** | `v3.3`
| **BC breaks**      | no
| **Tests pass**     | yes
| **Doc needed**     | no

Backport of https://github.com/ibexa/fieldtype-richtext/pull/118

Failing test should be fixed after https://github.com/ezsystems/ezplatform-richtext/pull/240.

**TODO**:
- [x] Implement feature.
- [x] Implement tests.
- [x] Fix new code according to Coding Standards (`$ composer fix-cs`).
- [x] Ask for Code Review.

### CODE REVIEW

File: src/bundle/Resources/config/fieldtype_services.yaml [hunk 1]
```diff
@@ -33,16 +33,16 @@ services:
...
-            - '%ezrichtext.style.namespace%'
-            - '%ezrichtext.embed.namespace%'
-            - '@?logger'
-            - '%ezplatform.ezrichtext.custom_tags%'
-            - '%ezplatform.ezrichtext.custom_styles%'
+            $repository: '@ezpublish.api.repository'
+            $configResolver: '@ezpublish.config.resolver'
+            $templateEngine: '@twig'
+            $permissionResolver: '@eZ\Publish\API\Repository\PermissionResolver'
+            $tagConfigurationNamespace: '%ezrichtext.tag.namespace%%'
```
Comment (konradoboza) [inline src/bundle/Resources/config/fieldtype_services.yaml]: Why double `%`?

Comment (barw4) [reply src/bundle/Resources/config/fieldtype_services.yaml]: miss copy-click

**Review by webhdx — APPROVED** [review]
I think it needs a rebase. CI issue is already fixed on the `main` branch.

**Review by konradoboza — APPROVED** [review]

**Review by alongosz — APPROVED** [review]

**Review by tomaszszopinski — APPROVED** [review]
QA approved on IbexaDXP commerce 3.3.

**Pull request merged**